src/validation.h

```diff
@@ -156,6 +156,7 @@ struct BlockHasher
 
 extern CScript COINBASE_FLAGS;
 extern CCriticalSection cs_main;
+extern CCriticalSection cs_LastBlockFile;
```

Comment:
I'd rather not expose more variables that are private to validation to the outside world.

Perhaps you can instead create a wrapper around `CalculateCurrentUsage`, which just grabs `cs_LastBlockFile` and calls the internal one, and expose that? That way the RPC code wouldn't need to know about the existence of that lock eve.

Reply:
Why not just lock `cs_LastBlockFile` in `CalculateCurrentUsage`? It's a recursive mutex and IMO should be locked where it's needed (BTW the same for remaining locks...).